src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -437,21 +438,21 @@ string relationshipEntity
+        [DataRow(new string[] { "id" }, null, new string[] { "num" }, null, "SampleEntity1", false, true, false,
             DisplayName = "LinkingSourceField is null, only ForeignKeyPair between LinkingObject and target is present. Invalid Case.")]
-        [DataRow(null, new string[] { "token_id" }, new string[] { "num" }, new string[] { "book_num" }, "SampleEntity1", false, true, false,
+        [DataRow(null, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity1", false, true, false,
             DisplayName = "SourceField is null, only ForeignKeyPair between LinkingObject and target is present.  Invalid Case.")]
-        [DataRow(new string[] { "id" }, new string[] { "token_id" }, new string[] { "num" }, null, "SampleEntity2", true, false, false,
+        [DataRow(new string[] { "id" }, null, new string[] { "num" }, null, "SampleEntity2", true, false, false,
             DisplayName = "LinkingTargetField is null, only ForeignKeyPair between LinkingObject and source is present. Invalid Case.")]
-        [DataRow(new string[] { "id" }, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity2", true, false, false,
+        [DataRow(null, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity2", true, false, false,
             DisplayName = "TargetField is null, , only ForeignKeyPair between LinkingObject and source is present. Invalid Case.")]
-        [DataRow(new string[] { "id" }, null, new string[] { "num" }, new string[] { "book_num" }, "SampleEntity1", true, false, true,
+        [DataRow(new string[] { "id" }, null, new string[] { "num" }, null, "SampleEntity1", true, true, true,
             DisplayName = "LinkingSourceField is null, only ForeignKeyPair between LinkingObject and target is present. Valid Case.")]
-        [DataRow(null, new string[] { "token_id" }, new string[] { "num" }, new string[] { "book_num" }, "SampleEntity1", true, false, true,
+        [DataRow(null, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity1", true, true, true,
             DisplayName = "SourceField is null, only ForeignKeyPair between LinkingObject and target is present. Valid Case.")]
-        [DataRow(new string[] { "id" }, new string[] { "token_id" }, new string[] { "num" }, null, "SampleEntity2", false, true, true,
+        [DataRow(new string[] { "id" }, null, new string[] { "num" }, null, "SampleEntity2", true, true, true,
             DisplayName = "LinkingTargetField is null, only ForeignKeyPair between LinkingObject and source is present. Valid Case.")]
-        [DataRow(new string[] { "id" }, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity2", false, true, true,
+        [DataRow(null, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity2", true, true, true,
```

Comment:
These display names aren't accurate because they don't align with your updates. They state "only ForeignKeyPair between" even when BOTH `isForeignKeyPairBetSourceAndLinkingObject` and `isForeignKeyPairBetTargetAndLinkingObject` are true.

With your new validations, can you clarify the purpose of this test by updating this test method's summary? 
The testmethod name is `TestRelationshipForCorrectPairingOfLinkingObjectWithSourceAndTarget` so a summary that answers:
1. what is a "correct pairing"
2. what is the purpose of declaring `isForeignKeyPairBet{Target/Source}AndLinkingObject` as true? 
    - why does false affect the test results?  
    - what effect do your new validations have on this?

Everything else looks good. Let's use this opportunity to make this test more readable for everyone.

Reply:
We've been discussing this offline, but I wanted to follow up with the information here in this PR. This test doesn't seem to be valid given the behavior we have decided on.

In every case, this test presumes that we have a state where we are missing some collection of fields, which according to our docs, and according to the behavior of the engine, is not valid. We can't have missing linkingSource/linkingTarget fields, and we can't have missing source/Target fields, nor can we have a case where all of that is missing. We must include those fields for the engine to be able to correctly form the relationships in our own necessary data models.

Because of that I think we should remove this test entirely, and then add back in the many to many validations that make sense in the follow up PR that covers the linking object being added to our engine's data model prior to validation of the config

Reply:
agreed